server/routerlicious/packages/lambdas/src/nexus/connect.ts

```diff
@@ -92,6 +93,9 @@ function composeConnectedMessage(
 		initialMessages: [],
 		initialSignals: [],
 		supportedVersions: ProtocolVersions,
+		supportedFeatures: {
```

Comment:
I curious what we can draw from ODSP experience as they already use "supportedFeatures"? Are there any end-of-life concerns if we decide to remove this feature at some point? 

Here is some from chat with Vlad:

[FluidFramework/common/lib/protocol-definitions/src/sockets.ts at f2b744a7f92e52fe40ec2e66bcb824bd3ac14cd9 · microsoft/FluidFramework (github.com)](https://github.com/microsoft/FluidFramework/blob/f2b744a7f92e52fe40ec2e66bcb824bd3ac14cd9/common/lib/protocol-definitions/src/sockets.ts#L70)
 
And same on line 164 for reverse direction.
 
This is where client sets these:
[FluidFramework/packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts at f2b744a7f92e52fe40ec2e66bcb824bd3ac14cd9 · microsoft/FluidFramework (github.com)](https://github.com/microsoft/FluidFramework/blob/f2b744a7f92e52fe40ec2e66bcb824bd3ac14cd9/packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts#L309)
 
And this is where we change client behavior based on what service told us:
[FluidFramework/packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts at f2b744a7f92e52fe40ec2e66bcb824bd3ac14cd9 · microsoft/FluidFramework (github.com)](https://github.com/microsoft/FluidFramework/blob/f2b744a7f92e52fe40ec2e66bcb824bd3ac14cd9/packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts#L442)

Reply:
ODSP have decided to use supportedFeatures for targeted signals as well.
https://github.com/microsoft/FluidFramework/blob/55b0f70003e598565d7813140a83e96b713c3d20/packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts#L302-L304

I have future client side changes that reflect theirs in this PR: 
https://github.com/microsoft/FluidFramework/pull/20331/files

I'm not sure about end-of-life concerns -- I assume that if the service responds saying they don't support a feature anymore the client code receives this message and respond accordingly.

Reply:
Maybe more of a questions that you can consider from client side - does it mean that client will need to support both variations (with and without targeting) forever?